src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -14,7 +14,8 @@ internal static class DocumentExtensions
     {
         public static bool ShouldHideAdvancedMembers(this Document document)
         {
-            return document.Options.GetOption(CompletionOptions.HideAdvancedMembers);
+            // Since we don't actually have a way to configure this per-document, we can fetch from the core workspace
+            return document.Project.Solution.Workspace.Options.GetOption(CompletionOptions.HideAdvancedMembers, document.Project.Language);
```

Comment:
@dpoeschl's understanding is correct @CyrusNajmabadi. When I originally did the split, I just moved everything I could to document options because "it was easy". Now that document options are marginally harder to get and potentially involves blocking, I've generally gone back to a model where on/off flags for features are just fetched from Workspace options, since we don't plan to expose editorconfig values for those.